### ISSUE TITLE
[Bug]: App theming feedback bugs

### ISSUE THREAD

**Issue by dilippitchika** [appsmithorg/appsmith#11533]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

Noting all the bugs from app theming's feedback here

[Rahul's feedback ](https://theappsmith.slack.com/archives/CGBPVEJ5C/p1646140579174489?thread_ts=1646039612.688579&cid=CGBPVEJ5C)
1. table buttons don’t have the option to reset, or the orange icon notifying that it is deviating from the theme
2. the input for select doesn’t have the same style items as normal

[Aswath's feedback](https://theappsmith.slack.com/archives/C02JV8G1MP0/p1646149069616919)
1. We can add any css we want inside properties using JS

[Nikhil's feedback] (https://theappsmith.slack.com/archives/CRVRBGU1K/p1646042115945239?thread_ts=1646039810.972839&cid=CRVRBGU1K)
1. Update the current theme to be hidden or prevent showing 2 themes when they are the same
2. The classic theme is being selected for all new apps instead of the default theme. This is a bug
3. The button group button color property control isn't showing the theme value
4. In checkbox, audio widget, filepicker etc. the color property is called background color. This is a misnomer. Can we fix these properties before they go out?
5. The map widget, multi-select, video widget don't round it's corners properly.

Issues list:
https://docs.google.com/spreadsheets/d/1rWKhq6wiwXogAwvWT0Y4o2lKdNitwIL29y9J5OYRtAk/edit#gid=0

### Steps To Reproduce

Follow the above slack threads.

### Environment

Deploy Preview

### Version

Cloud

**Comment by dilippitchika** [comment]
@jsartisan @keyurparalkar please keep adding more issues found from feedback here, just to make sure we keep resolving all these.

**Comment by jsartisan** [comment]
> We can add any CSS we want inside properties using JS

We won't be able to handle this right now because we will need validations for it. This can take time because I am not sure how will we handle validations here. for e.g - for background property - we will need to handle all possible values like `red`, `blue`, `#ff1233` and `url(path-to-image)`. 

cc: @dilippitchika

**Comment by dilippitchika** [comment]
> We won't be able to handle this right now because we will need validations for it. This can take time because I am not sure how will we handle validations here. for e.g - for background property - we will need to handle all possible values like red, blue, #ff1233 and url(path-to-image).

No issues on this 

> We did this intentionally. We decided to keep the green theme as the default for now till we do reskinning project. after that we will use the purple theme as the default one.

Then we should not call the purple theme "Default". We should edit the name to be different, my current suggestion - "Modern"

**Comment by shwetha-ramesh** [comment]
@jsartisan tested theming DP. And here are the few issues I have noticed

1. When you save a theme and later delete the theme, the pop up message displayed has wrong message on it. Message must read “Do you really want to **delete** this theme”. 'Delete' is missed out.
![image](https://user-images.githubusercontent.com/92293815/156549800-2512b01e-9348-4a78-988f-11f84873da74.png)

2. When user deletes a theme a pop message is displayed “Do you really want to delete this theme? This process cannot be undone” However when you click on delete a toast message is later displayed. “THEME abc deleted. UNDO”.
The user has an option to UNDO, but when clicked on it the deleted theme does not appear back. 
The pop up message displayed and the toast message has contradictory messages.
![image](https://user-images.githubusercontent.com/92293815/156550406-a06e3a02-629b-4bde-9962-442ec76fcfc9.png)

3.  Duplicate theme names can be saved without any changes made it to it.
https://loom.com/share/9ba342b2054f4d09bf4ee80fbee756db
![image](https://user-images.githubusercontent.com/92293815/156551728-62e14823-9539-4b2c-aad0-57fbd55bcf7c.png)

4. Tool tips can be elaborated rather than short form for Border radius and shadow
![image](https://user-images.githubusercontent.com/92293815/156551820-34665602-5783-4f30-9959-0d6bd7bf0858.png)

5. Different spelling for Color in same property
![image](https://user-images.githubusercontent.com/92293815/156551976-3392dfa1-c45f-48a4-b7c1-03fb863e3a80.png)

6. Button group widget appears as one big widget and there is no separator seen from one button to another
![image](https://user-images.githubusercontent.com/92293815/156552043-073fda0a-6ced-46e3-8932-b2a06abf2b9b.png)

7. No theme styling properties applied for divider, document viewer, image, Multiselect widget

8. For checkbox group widget when user adds additional options the checkbox are not seen even on resize of widget; only with scroll the widgets can be seen.
![image](https://user-images.githubusercontent.com/92293815/156552324-58b47b45-4bfc-483d-abe9-d276a502b398.png)
https://loom.com/share/7cc1762ba5ed41bdb5d8dec197e05e91

9.Theming not applied for File picker widget button and Modal.
![image](https://user-images.githubusercontent.com/92293815/156552453-0c34cd5f-bbbe-416e-ac67-28ccb3e26324.png)
![image](https://user-images.githubusercontent.com/92293815/156552586-0b6c0f3f-a6c4-4636-871e-d22175ea06bf.png)

10. Background color not applied for Multitreeselect widget drop down. Also a white cursor appears and text has white background. Also in Rounded theme text “no results found” is cut off
![image](https://user-images.githubusercontent.com/92293815/156552715-cfacb054-68b1-46a7-87a0-cc9774d61c6d.png)

11. For default image widget Border radius are not visible .
![image](https://user-images.githubusercontent.com/92293815/156552848-d4b6409a-9a78-4184-9dc1-c50a30d19781.png)

12. Multiselect widget does not have theming applied to it. Highlighting on drop down and border radius does not conform to theme selected for Multiselect widget. 
![image](https://user-images.githubusercontent.com/92293815/156553142-3c97110f-d995-4b24-a2c7-9bd0eef3e351.png)

13. Styling properties (radius and shadow) not visible for Radiogroup widget.
![image](https://user-images.githubusercontent.com/92293815/156553312-c98b7f03-531e-4571-8785-12fe95019ab5.png)

14. Styling properties (radius and shadow) not visible for Rating widget . Also theme not applied.
![image](https://user-images.githubusercontent.com/92293815/156553329-5a44e767-4488-4f19-a7f9-6c63b591fd8c.png)

15. Theming not applied for Table widget’s filter property
![image](https://user-images.githubusercontent.com/92293815/156553468-5205f982-92f2-423e-b3be-dd2db6999b88.png)

16. Styling not applied for Text widget
https://loom.com/share/f4330b00485e482f96c8b248eb6936da

17. Border radius not applied for top border of video widget  
![image](https://user-images.githubusercontent.com/92293815/156555586-7de77154-378f-45fa-bebb-fa72b066f52d.png)

18. Not able to place widgets on bottom of the canvas
https://loom.com/share/c10654f0b5364e3bacd5f98eccca30c9

19. Reflow not supported.
https://loom.com/share/ec7038d218fe4053ad5dec616177530d

20. For select widget, the drop down overflows from the widget
![image](https://user-images.githubusercontent.com/92293815/156556399-31373193-61e4-4617-a0c2-a55407601234.png)

**Comment by dilippitchika** [comment]
Other bugs: 

Very serious - 
https://www.loom.com/share/4d7c6735f8bd4267bbdd4f66ead97b89
![Screenshot 2022-03-03 at 5 54 35 PM](https://user-images.githubusercontent.com/19730984/156564581-c634463a-7ea9-4e36-9891-fb4d86b9e004.png)

Rest 
Saved theme shows up twice in the list
![Screenshot 2022-03-03 at 5 50 03 PM](https://user-images.githubusercontent.com/19730984/156564338-bc3d1545-e816-416f-b32a-872a48a19be1.png)

Use a save icon and not a download icon - people are getting confused
![Screenshot 2022-03-03 at 5 53 27 PM](https://user-images.githubusercontent.com/19730984/156564384-df476336-ba5a-448c-bede-90984d978c24.png)

Going to theme list and then selecting a widget and then de-selecting it still shows the theme panel
![Screenshot 2022-03-03 at 5 54 00 PM](https://user-images.githubusercontent.com/19730984/156564446-a3769f22-b1ba-48eb-bb27-80182bb69ef0.png)

Make the circle clickable in the color picker widget
![Screenshot 2022-03-03 at 5 54 10 PM](https://user-images.githubusercontent.com/19730984/156564528-428d2ef1-eddf-4cd4-8b6d-f92ac57a7334.png)

Global theme border radius doesnt change the containers inside a container

**Comment by shwetha-ramesh** [comment]
@jsartisan 
21. When long theme name is given the name abruptly cuts off
![image](https://user-images.githubusercontent.com/92293815/157170209-0e4bc2f3-0eff-4c12-aa45-d5931a1a0e09.png)

**Comment by keyurparalkar** [comment]
@shwetha-ramesh, Please find below the updates for the bugs:

> No theme styling properties applied for divider, document viewer, image, Multiselect widget

> For default image widget Border radius are not visible .

In the Image widget, the border-radius is applied to it’s container and not to it’s image.  When `Object fit = contain` it will try to fit the image inside the container so that aspect ratio remains the same that is why you were seeing half baked border radius. **Note:** The border radius still exists when `Object fit = contain`. 
Also, when `Object fit = cover` the theming properties exists.

<hr />

> Styling properties (radius and shadow) not visible for Radiogroup widget.

> Styling properties (radius and shadow) not visible for Rating widget . Also theme not applied.

For Rating and radio widget, we don't have any theming properties like `border-radius` or `box-shadow`. Only `colors` can be changed.

<hr />

> Styling not applied for Text widget

The `border-radius` and `box-shadow` property for the text widget is removed.

<hr />

> Border radius not applied for top border of video widget

The border radius still exists. It’s just that the user is not able to see it since the content of the video gets adjusted based on how we resize the video widget. Please refer to this slack thread for more information: https://theappsmith.slack.com/archives/C02JV8G1MP0/p1646381970834139
We have added background color property for the video widget so that it will help to distinguish the border-radius.

<hr />

> Reflow not supported.

Please enable the reflow by click the checkbox present in the property pane with label: “Turn on new drag & drop experience BETA”

**Comment by jsartisan** [comment]
For better tracking, we are trackng issues here - https://docs.google.com/spreadsheets/d/1rWKhq6wiwXogAwvWT0Y4o2lKdNitwIL29y9J5OYRtAk/edit#gid=0

**Comment by shwetha-ramesh** [comment]
@jsartisan observed one more wrt Pages
22. When a certain theme is applied to an application, subsequent pages must also have same theme applied, however currently in DP when a new page is created the theme is not applied immediately but only by switching back forth the theme gets applied. 
https://loom.com/share/d8c65ce6f4bf41e6a702e03334eb5023

**Comment by shwetha-ramesh** [comment]
@jsartisan @keyurparalkar  tested the above mentioned issues after fix and here are the status.

Issues reported by me 
1. Fixed

2. Fixed. Undo is removed from toast message.

3.  a) When user clicks on _Save theme_  and gives a duplicate name “Name must be unique” message appears, however when you click on cancel and click on Save theme again "Name must be unique" message is already displayed even for blank theme name. Expected behavior would be : When user enters duplicate name "Name must be unique" message must be displayed also 'Save Theme' button must be disabled; ideally the message should appear when same theme name is given and not for empty theme names. Also unless any characters are entered in theme name 'save theme' button must appear disabled. 
![image](https://user-images.githubusercontent.com/92293815/158108528-bfaaa4c1-043a-4e75-9a92-2f0939c7354f.png)
    b) Same theme settings with different names can be saved.
    c) Theme names accepts numbers and special character
![image](https://user-images.githubusercontent.com/92293815/158108472-c8db0ef9-63df-4d4a-a4e1-068fdef51cf9.png)

4. Still exists

5. Fixed

6. Separators are now seen inside the button group widget but the drop down for _More_ does not have fully rounded applied to it.
![image](https://user-images.githubusercontent.com/92293815/158108887-f17dfea8-e0eb-425f-9b3e-8c9142de4ce5.png)

7. Still exists
    a) Theming properties not seen for Divider widget. The divider color appears grey
    ![image](https://user-images.githubusercontent.com/92293815/158109192-926c6b2a-ca78-431a-bdc1-358fb611e5c8.png)
    b) Theming properties not seen for Document viewer widget.
    ![image](https://user-images.githubusercontent.com/92293815/158111193-d8bd2bb0-2332-44b2-ae6a-5c3c2bc1716d.png)
    c) Image widget - fixed
    d) Multiselect widget - Theming is now applied. However filter tray is not highlighted, in drop down the selected options do 
    not have highlight over selected options, on hover on checkbox is not highlighted, after you make changes in widget the revert option with orange dot is not present
    ![image](https://user-images.githubusercontent.com/92293815/158112709-2eab5a74-970b-498d-8bb8-cf9623debb0a.png)
    ![image](https://user-images.githubusercontent.com/92293815/158113023-3fd2b05c-e1cf-4aa7-a076-171ce05619e4.png)

8. Still exists. Widget spacing looks completely different from release. On theming when more options are added the vertical alignment from one another looks very compact. Also on hover the checkbox do not highlight.
![image](https://user-images.githubusercontent.com/92293815/158112401-72591863-af81-48bd-b4d4-bf20643d5188.png)

9. Still exists

10. For multitree select widget background color is now removed but the drop down is not fully rounded and search tray is not visible
![image](https://user-images.githubusercontent.com/92293815/158113386-dd332ac8-cb08-40e4-829c-ad948911f964.png)

11. Fixed

12. Multiselect widget now theming is applied. However filter tray is not highlighted, in drop down there is no highlight over selected options, on hover on checkbox is not highlighted, after you make changes in widget the revert option with orange dot is not present
https://loom.com/share/028779f49e054aa5810cdcf93b33492d

13. & 14. Addressed. widget does not consists of any theming properties except the color.

15. Still exists

16. The border-radius and box-shadow property for the text widget is now removed. However ```text size``` notations differ from release. Also no revert option for ```Font family```
![image](https://user-images.githubusercontent.com/92293815/158116414-4787dead-4e20-4716-b9f2-8c3f93faaa98.png)
![image](https://user-images.githubusercontent.com/92293815/158116152-611441ca-6e3a-48bc-97cb-aa9efb4e11be.png)

17. Background color property is added for video widget. However when a new url is added the rounded edges does not apply

18. Still exists

19. Corrected. Reflow is supported after checkbox selected.

20. Drop down overflows from the select widget is fixed in view mode but for long label names in deploy mode the drop down still overflows. But Box shadow is not seen for select widget.
Also when a long label name is given the widget flows out of canvas
![image](https://user-images.githubusercontent.com/92293815/158119615-c2fd4462-712b-4f63-a8ed-a2e4551a4319.png)
![image](https://user-images.githubusercontent.com/92293815/158119635-1c41011e-5459-49b6-87fe-3fbd741c1724.png)

21. Still exists

22. Still exists

**Comment by shwetha-ramesh** [comment]
New issue
24. Tabs widget ui is cut off
https://loom.com/share/cc3fb1efd97a4f44b8df4cf6e2bec6e8
![image](https://user-images.githubusercontent.com/92293815/158318617-5ff0532a-c8f2-44c5-98a5-dba9fba88fa3.png)

**Comment by shwetha-ramesh** [comment]
25. Date picker - box shadow does not have revert option
![image](https://user-images.githubusercontent.com/92293815/158542592-55bf6deb-5d88-4d0a-92e8-6ec608ce4bf6.png)

26. Select fully rounded theme, now drop a list widget on canvas, select the first tile of list widget - now change the border radius to none or md. The highlight is cut off.
![image](https://user-images.githubusercontent.com/92293815/158542511-2e0c553d-6b67-48d1-984f-2d9362e9dd2b.png)

**Comment by shwetha-ramesh** [comment]
@jsartisan @keyurparalkar 
29. Click on Save Theme, don’t give any name keep it blank click on Save theme button – see “Classic Copy” get saved. Repeat the same step – and see “Classic copy” saved again
![image](https://user-images.githubusercontent.com/92293815/159464556-5e171f42-2edc-4445-816c-7292fe22a00f.png)

30. Button group widget for vertical alignment is wrong in DP when more buttons are added.
![image](https://user-images.githubusercontent.com/92293815/159464621-36c19de0-7d1d-4350-a5e9-06d9bfb8d163.png)

**Comment by shwetha-ramesh** [comment]
@jsartisan below are the issues found in DP after fix and as mentioned in doc.

1. Step a: Click on Save theme > In Modal click on Cancel > Click on Save theme again > In modal now click on Save theme button > Theme gets saved successfully.
   Step b: Repeat above step once more > 'Classic Copy' gets saved twice

2. Save theme button can be disabled unless character input
![image](https://user-images.githubusercontent.com/92293815/159901747-ae3de550-ace8-4dcf-aa89-1108473d01b4.png)

3. a)Button group widget appears different in Edit mode and Deploy mode when Orientation is Vertical and widget is resized.. 
b) Also a lot of spacing is observed for drop down and the button
c) Hover is fixed
d) widget looks different only when button is present.
![image](https://user-images.githubusercontent.com/92293815/159908896-67b89003-c933-4a63-908f-2842dea7daf2.png)
![image](https://user-images.githubusercontent.com/92293815/159902252-71bc9058-ae66-4e9d-b149-418ee694ad4a.png)
![image](https://user-images.githubusercontent.com/92293815/159902553-b4d752f6-114f-4eab-8b7a-9087d95dd858.png)
https://loom.com/share/991b22a27c68466297d0b4071711d93a

4. No hover found for FONT in Theme properties
![image](https://user-images.githubusercontent.com/92293815/160057927-5ceb378d-15b8-4a4f-b195-4a0dd1cd9704.png)

5. Checkbox alignment left right isn't working
![image](https://user-images.githubusercontent.com/92293815/160062111-3e73c5bc-13c5-4c97-9bbd-f0ea022dd139.png)

6. For phone input widget on clicking on the input the whole widget is highlighted, it must highlight only the input part
![image](https://user-images.githubusercontent.com/92293815/160063148-306d7f57-2af0-4657-b9ca-bf747183f4d4.png)

7.Checkbox group widget in disabled state doesn't show tick mark for selected checkbox
![image](https://user-images.githubusercontent.com/92293815/160066729-733d2a54-9324-407b-b6f6-62c8bd96cf52.png)

8. In text widget last letter gets cut off when it is Italics and right aligned.
![image](https://user-images.githubusercontent.com/92293815/160077089-5819ff29-e5b9-4cfe-a2cf-021beb1801b4.png)

**Comment by shwetha-ramesh** [comment]
@keyurparalkar as discussed please find the below issues found while testing Migration on theming.
1. Box shadow selection is not shown
![image](https://user-images.githubusercontent.com/92293815/161050218-1d055593-ced1-40c4-8de9-879f0b6bfc60.png)

2. (#migration) For text widget when Heading 2 and Paragraph 2 is selected in theming JS is enabled by default and shows 1.125 and 0.75 respectively but when you toggle off JS it shows ‘base’
![image](https://user-images.githubusercontent.com/92293815/161050302-b87d30c1-5a51-465a-9e57-e426b1439385.png)

3. (DP issues) For border radius in Button widget when first option is selected, in theming no selection is shown and it shows empty when JS is toggled.
![image](https://user-images.githubusercontent.com/92293815/161050430-7ba8714f-4690-4e1d-a7f8-532366b4ffb5.png)

4. (#Migration) Button widget has bold border by default on migration
![image](https://user-images.githubusercontent.com/92293815/161050553-9d862948-06a7-4373-9d3d-52adc5f38ea4.png)

5. (#Migration ) For Container widget border radius does not show any unit, it just shows the number and not in pixels.
![image](https://user-images.githubusercontent.com/92293815/161050619-809fc6df-9310-4d2b-a910-6e75e99ec490.png)

6. (#Widget level) Border radius for none shows 0px, md = 0.375 rem, lg = 1.5 rem – no consistency in units

7. (#Widget level) Input widget icon appears bold
![image](https://user-images.githubusercontent.com/92293815/161050777-c6ce816d-0fda-49ce-bc66-b96d8b146ac0.png)

8. (#Migration) On migration the input widget does not highlight, border color is not migrating 
![image](https://user-images.githubusercontent.com/92293815/161050847-58c803b9-20ac-4490-bad2-6c32841c5f2d.png)

9. (#Migration) No revert option for border radius In multiselect widget
![image](https://user-images.githubusercontent.com/92293815/161050903-a4dcd4ea-b808-4907-b8b1-4251b6b0a94c.png)

10. (widget level) No revert option for Icon color in Audio recorder widget
![image](https://user-images.githubusercontent.com/92293815/161050962-8f2aa98f-1493-4a84-a34d-3cfd9172f392.png)

**Comment by shwetha-ramesh** [comment]
@keyurparalkar below are few more issues found while testing migration 
11.	Audio recorder widget shows 9999px for Border radius. Standard unit must be used across all widgets
![image](https://user-images.githubusercontent.com/92293815/161488571-9178f06d-64c3-4e59-97fd-dc47e2e92fbb.png)

12. No revert option found for select family
Select = No revert for Box Shadow
Multiselect = No revert for Border Radius, Box shadow
MultiTree select = No revert forBox shadow
Tree select = No revert for Box Shadow

13. On migration for Currency input and date picker widget by default the widget border, drop down highlight, date selection appears pink (theme selected color). 
![image](https://user-images.githubusercontent.com/92293815/161488795-1a7ac924-f73c-4b63-836b-a81859052eba.png)

14. For Form widget when there is no border radius mentioned on release, when migrated on DP it shows ‘undefinedpx’
![image](https://user-images.githubusercontent.com/92293815/161488866-17b9b75b-5eaa-48b3-8fa0-483ece8cada9.png)

**Comment by shwetha-ramesh** [comment]
@keyurparalkar i
15. Input widget shows theme border highlight on migration
![image](https://user-images.githubusercontent.com/92293815/161754600-d1f8bfa2-3587-4301-a6f0-aedbf5171d9a.png)
![image](https://user-images.githubusercontent.com/92293815/161754632-08918d57-7ad5-4231-a21e-2dc27d906819.png)

16. Theme is applied on List buttons on migration
![image](https://user-images.githubusercontent.com/92293815/161754679-6e771972-6bde-4979-8e85-f8d278e5b773.png)

17.	Box shadow is not applied to the whole widget but individual lists
![image](https://user-images.githubusercontent.com/92293815/161754755-07936c2b-574b-4f3a-96e8-4cd54837a094.png)

18. Menu button has green hover on migration and not grey.
![image](https://user-images.githubusercontent.com/92293815/161754822-41b9c149-ab0c-4d41-8d08-adc8aa1448a5.png)

**Comment by btsgh** [comment]
19. https://www.loom.com/share/063ee62c253c42c09213c5e5704b2669
Steps -
- Drag and drop say a button widget 
- (Theming is Default)  
- Click on Change Theme.  
- Choose Sharp theme and click on Apply Theme. 
- A message comes to the effect “Theme Sharp Applied.. Undo” 
- Click Ctrl-Z / Cmd-Z from keyboard to undo the theme.  
- The theme is undone, but the message on top still remains for a while-  “Theme Sharp Applied.. Undo” 
- Click on Undo on that message. 
- At this point, the Button widget itself gets undone. The canvas becomes blank.

**Comment by shwetha-ramesh** [comment]
@keyurparalkar #MigrationTestingIssues

19. In Modal on Migration the Close button text is overlapped when Icon is added.
![image](https://user-images.githubusercontent.com/92293815/162119031-d35fe9be-ebc6-4244-83e6-edeec8d88931.png)


20. . Multi tree select, Multi select has theme highlight on migration
![image](https://user-images.githubusercontent.com/92293815/162119009-076dcc3c-920f-4fc0-93e8-6a65e0903122.png)

21. Multi select has no revert option for box radius on migration.
![image](https://user-images.githubusercontent.com/92293815/162119082-be3905b7-73c7-4230-a9aa-6e77518b0335.png)

22. All select widgets, input widgets has drop down wrt theme on migration

**Comment by btsgh** [comment]
23. 
- Create a Form, add two input widgets and on Select widget.  
- On the Submit button, bind it to a query (Query1) such that it updates an excel data source.
- Apply a theme other than default  
- Change font to Montserrat or Ubuntu or any font 
- Deploy the app
- The font is seen as applied to the header region as well - ("<Appname>" … "Share" / "Edit App") and also on the bottom right "Built on appsmith_" text.  The font changes should only be applied on the canvas.
<img width="1350" alt="TDPFontChanges1" src="https://user-images.githubusercontent.com/101863839/162170230-b447554b-4119-40ec-8d4e-a310c657f91e.png">
<img width="1067" alt="TDPFontChanges2" src="https://user-images.githubusercontent.com/101863839/162170260-d46de6cf-9308-4907-961b-8a38b22a4203.png">

**Comment by btsgh** [comment]
24. "Built on appsmith_" on the bottom right corner of deployed screen - If I change background colour on a theme, the "Built on appsmith_" background colour also gets impacted. 
Since its a logo, the changes to canvas should not get reflected here. so background color should not reflect on that logo"
<img width="1022" alt="TDP_BuiltonAppsmith" src="https://user-images.githubusercontent.com/101863839/162173422-fbf8db29-bdd1-477f-9a8b-a1ee11d58bff.png">

**Comment by shwetha-ramesh** [comment]
@keyurparalkar #MigrationTestingIssues
25. Page breaks when Progress widget is opened on theming DP

26. No border radius for radio group widget.

27. On migration Rating app changes it stars higlight without hover on the stars.
       https://loom.com/share/560e89b97197435c810a278c7d92d15f

28. Change the font size in release on RTE widget > Now in appsmith check it is changed back to Paragraph

29. No scroll observed on deploy mode on theming, also a white cast is present at the end.
![image](https://user-images.githubusercontent.com/92293815/162175072-0052b8e7-c180-46fb-b4ab-b6825de25b49.png)

30. Drag and drop icon button widget on migration click on revert option for button color > now toggle the JS twice and observe the color disappears on the widget and the field
![image](https://user-images.githubusercontent.com/92293815/162178055-46eed7c7-7c96-4d09-ab62-cc781813e1a5.png)

**Comment by btsgh** [comment]
31.
1. In the COLOR section, Click on primary colour circle - change colour
2. Now click again on primary colour to modify the colour 
3. The Primary colour pop up does not come up and the text area below it to enter the colour hex value also gets closed.
4. We need to click on primary colour circle again to see the colour box.
5. Same is the behaviour with background colour also. 
Expected behaviour - on step 2, the primary colour should come up and the text area to enter hex value should also remain.
<img width="287" alt="Colorsection" src="https://user-images.githubusercontent.com/101863839/162182568-a68b5fb1-326e-4dc1-a2f0-6ee5b77cab51.png">

**Comment by shwetha-ramesh** [comment]
31. Blue highlight appears on change for Switch group widget on migration.
https://loom.com/share/1f62e44e2c6b460c97eddf2a3801a499

**Comment by btsgh** [comment]
35.

**Comment by btsgh** [comment]
35.
- Checkbox Group Widget .. Drag and drop checkbox group widget on Theming DP
- Drag and drop Drag and drop checkbox group widget on Release
- Spacing in Theming looks more crushed than a neater view in release. 
- In this case, on theming when more options are added, the newer options are tried to be fit into this UI, and text look cut off.
- In Release, these options are arranged neatly such that there is no text cutoff, and user needs to scroll down to see them.
- Since Theming is all about UI look and feel, it would be nice if the spacing in Theming DP is maintained similar to Release.

This was called out in Point 13 comments. Re-raising to see if this can be fixed.
<img width="587" alt="DCheckboxGroupWidget_Release_AfterAddingOptions" src="https://user-images.githubusercontent.com/101863839/162200080-b9e2c9d0-15df-4480-884e-b7c78a8d3114.png">
<img width="566" alt="DCheckboxGroupWidget_TDP_AfterAddingOptions" src="https://user-images.githubusercontent.com/101863839/162200100-266d3b4d-3693-40b3-a63e-9891b96af0dd.png">
<img width="339" alt="DefaultCheckboxGroupWidget_Release" src="https://user-images.githubusercontent.com/101863839/162200103-dc95c233-e02c-4dfe-b9c1-48d052f258e9.png">
<img width="414" alt="DefaultCheckboxGroupWidget_TDP" src="https://user-images.githubusercontent.com/101863839/162200106-dd6e633f-a3be-4480-85e6-8722441438df.png">

**Comment by btsgh** [comment]
@keyurparalkar #MigrationTestingIssues
38. 
-In theming DP, change theme to rounded, add your own custom colours and save it as MyTheme4. 
-Now in release url, create a page with Checkbox widget. 
-Open this page on Theming DP. 
-MyTheme4 is not applied completely. 
-Instead of the checkbox being rounded and shaded with a background of dark blue as per MyTheme4, the checkbox is square with a green background
<img width="325" alt="Chkbox_DirectApp" src="https://user-images.githubusercontent.com/101863839/162220359-aa49aa1d-6aef-403e-8d41-b719b1609fac.png">
<img width="457" alt="Chkbox_MigratedApp" src="https://user-images.githubusercontent.com/101863839/162220363-daecf488-61e9-413f-8698-10d15c85c983.png">

**Comment by btsgh** [comment]
@keyurparalkar #MigrationTestingIssues
In theming DP, change theme to rounded, add your own custom colours and save it as MyTheme4. Now in release url, create a page with Tree Select widget. Open this page on Theming DP. MyTheme4 is not applied correctly. The issues are - 
-The Hover highlights are grey instead of light blue as per the theme. 
-The edges are sharp instead of rounded. 
-The default selected radio button is not visible in the dropdown
-On selection of any option in the dropdown, corresponding radio button disappears
<img width="399" alt="TreeSelect_TDP_Direct" src="https://user-images.githubusercontent.com/101863839/162241515-9b496dec-dd49-4e4c-82e6-5b8ac7d9daab.png">
<img width="397" alt="TreeSelected_TDP_Migrated" src="https://user-images.githubusercontent.com/101863839/162241545-0ebe14bb-0609-49ca-9a6d-e39d01c7a999.png">

**Comment by btsgh** [comment]
40. Drag and Drop modal widget onto the canvas. Confirm button "m" is getting cutoff on Modal widget
<img width="589" alt="Modal_Release" src="https://user-images.githubusercontent.com/101863839/162246603-301debc5-4697-45ca-a17b-ecd95bf3c8cd.png">
<img width="605" alt="Modal_TDP" src="https://user-images.githubusercontent.com/101863839/162246622-6b3be42c-377b-44c1-8810-ca2afe19c48b.png">

**Comment by btsgh** [comment]
41. Add a Select widget to the Canvas. Click on the dropdown .. issues found - 
- Hover over the options in the dropdown is grey, does not take the theme colour
- The filter text box is fully rounded, as against the partially rounded for other select drop-downs Filter text box
- The Select option area, on click, is highlighted in Grey instead of highlighting in the selected theme colours"
-
<img width="301" alt="SselectWidget_TDP" src="https://user-images.githubusercontent.com/101863839/162252623-103ae469-f621-40ad-bf19-990048e2ac6e.png">

**Comment by btsgh** [comment]
42. Video widget -  On applying a background colour, and then undoing it, (cmd + z), the colour goes away from the background. However, the hex value remains in the background colour field.
<img width="321" alt="TDP_BGColor1" src="https://user-images.githubusercontent.com/101863839/162389270-038c8bd7-4a67-434d-a71a-b6ebf8d6b3e6.png">
<img width="735" alt="TDP_BGColor2" src="https://user-images.githubusercontent.com/101863839/162389280-dfa3847e-51bd-494b-994b-9d82cd3cd939.png">

**Comment by btsgh** [comment]
43. In Tree select widget, click on the dropdown, Enter some text in the filter field - the close (x) icon appears. Move the mouse over the x, see that the grey background on the x symbol is cutoff

**Comment by shwetha-ramesh** [comment]
@btsgh This is observed in all Select widgets.

**Comment by btsgh** [comment]
44.
1. Create a Switch Group Widget
2. Choose OnSelectionChange -> Show Message: ""Selected"" -> Type: Info
3. Click on Deploy
4. Choose an option. 
5. The message ""Selected"" comes up on the top of the UI (Near the URL of the page)
As per release behaviour, the message ""Selected"" should come up on the right side of the UI

<img width="1233" alt="Message_TDP" src="https://user-images.githubusercontent.com/101863839/162678262-09164ce6-f8db-4d20-a921-48ab77495a60.png">

<img width="1272" alt="Message_Release" src="https://user-images.githubusercontent.com/101863839/162678245-e2923f02-f9b4-49f5-bbe7-82149726be51.png">

**Comment by btsgh** [comment]
45.
"Table widget - Add a table widget in a page in release url, 
- Add cell background colour - #FFC13D
- Add text colour - F86A2B
- Set text size as Paragraph 14px
- Make text align to center
- Make vertical alignment to down
- Rearrange the columns. Open this page in the theming dp
>We can see that the background colour does not apply completely on the three dots for truncate text
>The text in the task column is cutoff due to center 
<img width="795" alt="Migration_TableWidget_bcgrd" src="https://user-images.githubusercontent.com/101863839/162681823-57ff9385-8daf-40d7-8021-60829df05594.png">
alignment."

**Comment by btsgh** [comment]
46. 
- Created a theme ""MyTheme4"" Which is rounded with additional background colours
- Drop an icon button widget .
- Increase the size of the widget and note the borders of the icon button. 
- It is square with rounded borders. 
- In Release, the round shape is retained.
<img width="420" alt="Icon_Release" src="https://user-images.githubusercontent.com/101863839/162731124-e00c5f35-711f-4ca4-9937-f6b49a22fe26.png">
<img width="209" alt="Icon_TDP1" src="https://user-images.githubusercontent.com/101863839/162731139-3de5c525-0245-42d3-99f3-4cddcfaf58cc.png">
<img width="963" alt="Icon_TDP2" src="https://user-images.githubusercontent.com/101863839/162731143-508314da-25b4-4b09-a7a6-f24d409deeab.png">

**Comment by btsgh** [comment]
47. On migration, the widget text fonts are not retained as per the app in the release. They change to the theming font. (For the theme that i have chosen, Monserrat is the font type that i have selected)
<img width="318" alt="SwitchGrpW_Release" src="https://user-images.githubusercontent.com/101863839/162894814-134d94e9-747c-4c27-af8d-97f65df38790.png">
<img width="383" alt="SwitchGrpW_TDP" src="https://user-images.githubusercontent.com/101863839/162894853-d95341d3-f655-4527-b415-81a5cd65a8ce.png">

**Comment by btsgh** [comment]
49. On a table widget, On binding to a google sheets query and pulling data, if the text in a column exceeds the column width, at some places, it is truncated properly with three dots, and on hover - full text is shown. (which is the right behaviour) ... but in some cases, the text is cutoff and no hover text is displayed. 
<img width="644" alt="TableWidget_TDP_ColumnTexttruncate_Incorrectbehavior" src="https://user-images.githubusercontent.com/101863839/162926374-b4b115a4-83d2-4629-abd7-714e76245599.png">
<img width="640" alt="TableWidget_TDP_ColumnTexttruncate_Rightbehavior" src="https://user-images.githubusercontent.com/101863839/162926388-5949958f-e53b-40e2-a029-4624d993bc48.png">

**Comment by shwetha-ramesh** [comment]
@jsartisan #DPIssues
50. Widget with drop down have different behavior for different family of widgets- ex: for fully rounded theme; Button group widget has sharp border radius drop down, for input family widget the drop down is fully rounded and for select widget family the drop down is medium rounded
![image](https://user-images.githubusercontent.com/92293815/162958539-b57688ad-5637-4000-b725-001e4142b0cc.png)

51. For select widget the drop down appears medium rounded in edit mode and sharp border radius on Deploy mode. 1
![image](https://user-images.githubusercontent.com/92293815/162960104-8f9410c0-bba8-4bd6-8c87-28ad1d5ada4a.png)

52. Also a sporadic behavior is observed for select widget on deploy mode where theme is not applied for select widget drop down
![image](https://user-images.githubusercontent.com/92293815/162960263-dfbe96ad-919a-4ead-95c7-44e5c9cb4328.png)

**Comment by btsgh** [comment]
56. 

-Drag and drop a table widget onto the theming dp canvas.
-In the properties pane, add two new columns (CustomColumn1 and CustomColumn2)
-Click on the settings of CustomColumn1. Choose ""Menu Button"" from Column Type dropdown
-Scroll down to LabelColor in the property pane. 
-Set any LabelColour from the palette. 
-Nothing happens on the button 
Expected - The label colour should get changed to what is selected in the palette.

<img width="642" alt="Tablewidget_TDP_LabelColour" src="https://user-images.githubusercontent.com/101863839/162995595-a4d67d04-d2d6-494d-9853-2ef5dce725c2.png">

**Comment by btsgh** [comment]
57. 
-Drag and drop a table widget onto the theming dp canvas.
-In the properties pane, add two new columns (CustomColumn1 and CustomColumn2)
-Click on the settings of CustomColumn1. Choose ""Menu Button"" from Column Type dropdown
-Scroll down to Menu Items, and click Add a New Menu Item. Add 2 menu items.
-Click on the OpenMenu button in the CustomColumn1 on the table.
-The dropdown that appears to show the Menu items is not having the Theme applied. 
<img width="1007" alt="TableWidget_CustomColumn_AddMenuItems" src="https://user-images.githubusercontent.com/101863839/163018050-82b60ca0-7967-45bb-9582-26673a911876.png">

**Comment by shwetha-ramesh** [comment]
@jsartisan #DPIssues for Table widget
59. Theme is not applied for hover on deploy mode
![image](https://user-images.githubusercontent.com/92293815/163106073-d320d1ec-cb27-4671-a3d0-4efa464bd14a.png)

**Comment by btsgh** [comment]
62. For fully rounded theme, add a button group widget. The More button is cutoff. 
<img width="385" alt="ButtonGroup_MoreCutoff__Secondarybutton_RoundedTheme" src="https://user-images.githubusercontent.com/101863839/163113262-90af64f6-4cbf-4739-a968-273469d796a6.png">
<img width="342" alt="ButtonGroup_MoreCutoff_RoundedTheme" src="https://user-images.githubusercontent.com/101863839/163113269-202586fa-5962-4d19-8470-74613228b3f1.png">

**Comment by chandannkumar** [comment]
63. JSON form's **Reset button** is not fit with Rounded theme whereas Submit button fits.
> Applied background color in JSON form to observe the issue
![image](https://user-images.githubusercontent.com/95341241/163114308-4b78b214-d9c5-4d9a-b6c4-334e1223d396.png)

**Comment by btsgh** [comment]
64. Drag and drop a Button Group Widget onto the canvas.
Click Settings of the More button on the Property Pane
Click on the Settings of First Option
Here, in the Style section, set a Background Colour and also set a Text Colour (Say background is dark green and text is chrome yellow
The colours get applied and their hex values are shown in the fields
Click on the back arrow on top to go to the More Property pane. 
Click on the First Option Settings once again. 
The text colour field shows empty. 
If we click on the More dropdown in the canvas, the background and text colours are maintained, but the text colour field shows up empty. This is incorrect behaviour"

**Comment by chandannkumar** [comment]
65. **App Font** box's UI is overlapping the border of properties present in global theming setting page
![image](https://user-images.githubusercontent.com/95341241/163119474-d24c866e-670d-47c9-abc1-2e33cff86940.png)

**Comment by btsgh** [comment]
66. Drag and drop a Button Group Widget onto the canvas.
Select Orientation as Vertical
Click on the Settings of the More Button
Change the Button Colour to anything from the palette
Once that is done, There is no revert option here to get back to theme colour. "

**Comment by shwetha-ramesh** [comment]
67. No revert option found for table widget column type

**Comment by btsgh** [comment]
71.
Drag and drop a Button Group Widget onto the canvas.
Select Orientation as Vertical
In the properties pane, change the box shadow property from default to some other box shadow. 
Once that is done, There is no revert option to get back to the theme's box shadow
<img width="589" alt="buttongroup_boxshadow_norevert" src="https://user-images.githubusercontent.com/101863839/163159452-66cd1b40-2f1f-4f2a-9485-b45e2fff1ecd.png">

**Comment by shwetha-ramesh** [comment]
72. Pagination control on table widget doesnt conform to theme color
![image](https://user-images.githubusercontent.com/92293815/163172456-a5f49a41-0394-4a4a-b6f4-f188d62b6c20.png)

**Comment by btsgh** [comment]
73. 
Drag and drop a table widget onto the theming dp canvas.
In the properties pane, go to the Steps column settings
Set a Text Colour from the property pane to say brown from the palette. Click out of the text box and do a cmd + z. The text colour does get reverted, but the text box still remains showing brown hex. 
Similarly set a Cell Background Colour. Click out of the text box and do a cmd + z. The cell background colour does get reverted, but the text box in the property pane still remains showing brown hex. 

<img width="1008" alt="TDP_TableWidget_TextandCellbgColor" src="https://user-images.githubusercontent.com/101863839/163188965-8a46e718-9f1d-494b-81b1-b5a2e01935d8.png">

**Comment by shwetha-ramesh** [comment]
74. When the column type is video in table widget the video opens only half 
![image](https://user-images.githubusercontent.com/92293815/163207411-1bf6c47b-efdb-4cde-b532-a0578aae70c1.png)

**Comment by chandannkumar** [comment]
75. Text Size is not getting updated on migration - Json form @keyurparalkar 
Release env:
![Screenshot (64)](https://user-images.githubusercontent.com/95341241/163945339-5bf79061-645e-411f-b810-581ad402fb63.png)

Theming DP:
![Screenshot (65)](https://user-images.githubusercontent.com/95341241/163945436-ee1ae3e6-f22a-4375-9156-b2beb3717e2e.png)

**Comment by chandannkumar** [comment]
78. Border Radius, Box Shadow fields are blank at first until clicked once on migration - Json form
![Screenshot (66)](https://user-images.githubusercontent.com/95341241/163947009-957830f2-b6ae-44fa-b007-1d0937d4e42c.png)

**Observation**: configuring box shadow and shadow color in same field seems complicated on theming
![image](https://user-images.githubusercontent.com/95341241/163947573-9e4098ae-e87e-44cc-a185-7f7fac6d8083.png)

**Comment by shwetha-ramesh** [comment]
#ButtonGroupMigrationIssues
79. Size of the drop down for Button group widget varies on Migration
![image](https://user-images.githubusercontent.com/92293815/163949025-189493f9-b4d8-40c5-a3d0-172c1ee668c6.png)

**Comment by btsgh** [comment]
On Release DP - Drop a table widget.
On Theming DP, with the default classic theme, open the above page. 
Change the theme now to MyTheme4 - (Rounded, Pink and Blue colours, Montserrat Font) 
There is a revert option for the Table border . Click that and it becomes Rounded
Now click on the settings button of Action button - 
Issues: 
Button Colour does not have a revert option .
Box Shadow is having the JS option enabled and is showing up empty

<img width="1109" alt="TW_Migration_Actionsbutton_NoRevert" src="https://user-images.githubusercontent.com/101863839/163987693-fc0c4555-718b-4253-8b7d-666ea57731a1.png">

**Comment by btsgh** [comment]
On Deploy, the header that comes up on the Deploy page is different from what is seen in the Release URL
<img width="1434" alt="THeader_Release" src="https://user-images.githubusercontent.com/101863839/163995762-090bd61d-94bd-4e21-85ac-628197e9482a.png">
<img width="1425" alt="THeader_TDP" src="https://user-images.githubusercontent.com/101863839/163995771-097d93bd-811e-46d0-b9c5-3b052d1379d7.png">

**Comment by btsgh** [comment]
Drag and drop a table widget in a page in Theming DP (Classic Theme is applied by Default)
Go to the Action button settings for Action column
Choose Button Variant as Secondary - Compare button with Release behavior - It looks different. Has a darker border as compared to Release
Choose Button Variant as Tertiary - Compare button with Release behaviour - It looks different. Has a border, while Release has no border.

<img width="1122" alt="Button_SecondaryVariant_Release" src="https://user-images.githubusercontent.com/101863839/164017028-f5ff5f94-2968-49c9-8289-bcc8209e1eee.png">

<img width="1068" alt="Button_SecondaryVariant_TDP" src="https://user-images.githubusercontent.com/101863839/164017056-3208518a-f7a9-43d5-884a-bd891343fa3c.png">

<img width="1120" alt="Button_TertiaryVariant_Release" src="https://user-images.githubusercontent.com/101863839/164017063-8bbc1536-e4b3-4268-9b55-76451e4039ea.png">

<img width="1049" alt="Button_TertiaryVariant_TDP" src="https://user-images.githubusercontent.com/101863839/164017070-4a8c863e-d1c8-4040-a2a7-58fdabd6da8f.png">

**Comment by btsgh** [comment]
On Release DP - Drop a table widget.
For the onRowSelected property, choose Show Modal - New Modal - Modal1
On Theming DP, with the default classic theme, open the above page. 
Change the theme now to MyTheme4 - (Rounded, Pink and Blue colours, Montserrat Font) 
Click on a Row. The Modal opens up
Open the Modal property pane
The Border Radius property has a revert option, and on doing that, the Modal border radius gets rounded. 
**Issues:
Theming colours and border radius is not applied for the Modal buttons (Close and Confirm)
Confirm button is cutoff.** 

<img width="810" alt="Modal_Migration_TW_ThemingDP" src="https://user-images.githubusercontent.com/101863839/164022566-d62babf7-8667-460f-bd21-ac522a91ce3b.png">

**Comment by btsgh** [comment]
On Theming DP, with classic theme, create a page and drop any widget 
Now click on deploy and do not move the mouse pointer from the deploy button area. 
Once the page is loaded, the mouse pointer is by default on the Edit App button
Now move the mouse away from the Edit App button
Observe the Edit App button is not visible - it greys out.

<img width="1415" alt="TDP_EditAppOnDeploy" src="https://user-images.githubusercontent.com/101863839/164157751-57a642e4-fd62-4505-804d-54bb72ffd734.png">

**Comment by btsgh** [comment]
In tree select widget - on changing theme, 
1. Theme is not getting applied on hover and font on the dropdown options
2. As soon as the dropdown is clicked, the dropdown UI shifts slightly to the left.

https://www.loom.com/share/f9c971fd76c249ba92497434161d7972

**Comment by chandannkumar** [comment]
#onMigration
Border width remains **0 or blank** by default and hence it is not visible after migration
![Screenshot (69)](https://user-images.githubusercontent.com/95341241/164235013-df322cb4-c3e9-49fb-a932-ae2f3bbdcb2b.png)
![Screenshot (68)](https://user-images.githubusercontent.com/95341241/164235054-1d97de29-6797-46d5-b972-2a6721e5471a.png)

**Comment by shwetha-ramesh** [comment]
#MigrationIssues
For button widget when box shadow color is applied it is JS toggled and shows the respective mapping with rgba code but when JS is off there is no selection seen.
![image](https://user-images.githubusercontent.com/92293815/164383454-274a2b25-4b39-4333-bcbd-7ad39825b437.png)

**Comment by btsgh** [comment]
For these widgets - DatePicker, Rich Text Editor, Checkbox Group, RadioGroup, Switch Group -
Label Text Size property change does not impact the label text

These widgets still point to the Release type Label Text Size options. (Heading 1 Heading 2 etc) as against the theming DP label text size options - (sm, base, lg, xl.. )

<img width="279" alt="TDP_LabelTextSize_OldFormat" src="https://user-images.githubusercontent.com/101863839/164423662-5e8b563c-f255-4e41-ba5c-b1823aac590e.png">

**Comment by shwetha-ramesh** [comment]
#MigrationIssues
List widget page number style varies
![image](https://user-images.githubusercontent.com/92293815/164428739-2ef42b98-ebf3-4d39-9ca4-df7d49c8aa65.png)

**Comment by shwetha-ramesh** [comment]
#Migrationissues
As per the observation for ButtonGroup Menu items the hover remain grey on migration where as for Menu button it appears green. This is inconsistent behavior
![image](https://user-images.githubusercontent.com/92293815/164430888-283b9136-f03d-4043-afa5-c5cf6123c007.png)

**Comment by shwetha-ramesh** [comment]
#MigrationIssues
For progress bar widget. Drop a progress bar on release. Move to theming DP open the same application. Now click on revert for Fill Color click on JS toggle twice and notice the color becomes empty
https://www.loom.com/share/f4b68cb12b8a466d8223fc457c4e03b3

**Comment by chandannkumar** [comment]
UI alignment is improper when size varies of the label name in Checkbox field type & Switch field type
![image](https://user-images.githubusercontent.com/95341241/164475040-a90ac7eb-63f1-41db-afc9-73aa96cdce99.png)
![image](https://user-images.githubusercontent.com/95341241/164475214-039275c5-1556-4e88-98cf-d091cda734a3.png)
